usr/share/rear/conf/default.conf

```diff
@@ -3289,37 +3289,44 @@ SIMPLIFY_TEAMING=no
 
 ####
 # Serial console support for the ReaR rescue/recovery system:
-# By default serial consoles get enabled if serial devices are found and
-# then matching kernel command line parameters like 'console=ttyS0,9600 console=ttyS1,9600'
-# are set when booting the rescue/recovery system (see KERNEL_CMDLINE above).
 # IA64 platforms require it and people need it when no VGA console is available.
+# By default there is serial console support in the ReaR recovery system
+# when 'getty' or 'agetty' and 'stty' can be found,
+# otherwise USE_SERIAL_CONSOLE is automatically set to 'no'.
+# When 'getty' or 'agetty' and 'stty' can be found
+# and there is a 'console=...' option in /proc/cmdline,
+# then USE_SERIAL_CONSOLE is automatically set to 'yes'.
 # Specify 'yes' or 'no' or leave it empty to use ReaR's automatism:
 USE_SERIAL_CONSOLE=
-# If you do not want to use all serial devices found as console you can specify the ones to use
-# e.g. SERIAL_CONSOLE_DEVICES="/dev/ttyS0 /dev/ttyS1" provided USE_SERIAL_CONSOLE is turned on.
-# By default (when empty) all serial devices found are used:
+# Which serial devices should be used for serial consoles
+# provided USE_SERIAL_CONSOLE is not set to 'no':
+# E.g. SERIAL_CONSOLE_DEVICES="/dev/ttyS0 /dev/ttyS1"
+# By default (when empty) the devices of the 'console=...' options in /proc/cmdline
+# that exist as /dev/ttyS* or /dev/hvsi* character device nodes are used:
```

Comment:
What if the kernel cmdline contains something like `console=ttyS1,9600 console=tty0` and SERIAL_CONSOLE_DEVICES is empty (the default)?
By the rule 
> the devices of the 'console=...' options in /proc/cmdline
> that exist as /dev/ttyS* or /dev/hvsi* character device nodes are used

it seems that only /dev/ttyS1 gets used (because tty0 does not match /dev/ttyS* nor /dev/hvsi* ) and the resulting console parameter for the rescue system will be just `console=ttyS1,9600`. But this seems to conflict with the stated intent `copy all the console= kernel arguments from the running system` in the PR description (because not all the console= kernel arguments will be copied).

Reply:
When 'getty' or 'agetty' and 'stty' can be found
and there is a 'console=...' option in /proc/cmdline,
then all 'console=...' options get copied
for the kernel of the ReaR recovery system
via COPY_KERNEL_PARAMETERS+=( console )
see prep/GNU/Linux/200_include_serial_console.sh

Reply:
This is a reasonable behavior. It is hard to see from the rules described in default.conf comments why it works this, though.
> By default (when empty) the devices of the 'console=...' options in /proc/cmdline
 that exist as /dev/ttyS* or /dev/hvsi* character device nodes are used
 (which excludes /dev/tty0 when there is 'console=tty0' in /proc/cmdline):

this sounds that the example above (`console=ttyS1,9600 console=tty0`) SERIAL_CONSOLE_DEVICES gets set to `/dev/ttyS1` only and `/dev/tty0` is excluded.

> When SERIAL_CONSOLE_DEVICES_KERNEL is empty but SERIAL_CONSOLE_DEVICES is specified
then the specified SERIAL_CONSOLE_DEVICES are used for the kernel:

this sounds that only `/dev/ttyS1` will then be used as the `console=` device, not `/dev/tty0`. But actually, according to your comment above, both will be used, which seems to contradict the description.

Maybe the lines
> By default (when empty) the devices of the 'console=...' options in /proc/cmdline
 that exist as /dev/ttyS* or /dev/hvsi* character device nodes are used
 (which excludes /dev/tty0 when there is 'console=tty0' in /proc/cmdline):

apply only to SERIAL_CONSOLE_DEVICE_SYSLINUX and SERIAL_CONSOLE_DEVICE_GRUB, but not to SERIAL_CONSOLE_DEVICE itself. (i.e. only `console=` arguments matching /dev/ttyS* or /dev/hvsi* will be used for the bootloader, but all arguments will be used for the kernel)? In that case, the description should be moved to the bootloader part?

Reply:
Grmpf!
Yesterday I had added three comments to that conversation
but now those seem to be lost.
Or did I add my comments to yet another conversation
that is now completely lost?
Cf.
https://github.com/rear/rear/pull/2961#issuecomment-1758968386

Reply:
@jsmeix have you found the comments now, in the other thread (i. e. they did not belong to this conversation)? Or are there also some comments in this conversation missing?

Reply:
Yes I found the comments.
I missed the ones that you found and posted here
https://github.com/rear/rear/pull/2961#discussion_r1358093014